src/bin/vhost_user_fs.rs

```diff
@@ -172,7 +183,23 @@ impl<F: FileSystem + Send + Sync + 'static> VhostUserBackend for VhostUserFsBack
             REQ_QUEUE_EVENT => {
                 debug!("REQ_QUEUE_EVENT");
                 let mut vring = vrings[1].write().unwrap();
-                self.process_queue(&mut vring)?;
+                if self.event_idx {
+                    // vm-virtio's Queue implementation only checks avail_index
+                    // once, so to properly support EVENT_IDX we need to keep
+                    // calling process_queue() until it stops finding new
+                    // requests on the queue.
+                    loop {
+                        vring
+                            .mut_queue()
+                            .update_avail_event(self.mem.as_ref().unwrap());
+                        if !self.process_queue(&mut vring).unwrap() {
```

Comment:
Could you please make sure you're not breaking the error propagation by not using `unwrap()` functions?

Reply:
Fixed, thanks. And, while I was there, I also removed the unwraps from Reader/Writer.